src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Input/RelayCommand{T}.cs

```diff
@@ -0,0 +1,191 @@
+
+using System.Runtime.CompilerServices;
+
+namespace System.Windows.Input;
+
+/// <summary>
+/// A generic command whose sole purpose is to relay its functionality to other
+/// objects by invoking delegates. The default return value for the CanExecute
+/// method is <see langword="true"/>. This class allows you to accept command parameters
+/// in the <see cref="Execute(T)"/> and <see cref="CanExecute(T)"/> callback methods.
+/// </summary>
+/// <typeparam name="T">The type of parameter being passed as input to the callbacks.</typeparam>
+public class RelayCommand<T> : IRelayCommand<T>
+{
+    /// <summary>
+    /// The <see cref="Action"/> to invoke when <see cref="Execute(T)"/> is used.
+    /// </summary>
+    private readonly Action<T> _execute;
+
+    /// <summary>
+    /// The optional action to invoke when <see cref="CanExecute(T)"/> is used.
+    /// </summary>
+    private readonly Predicate<T> _canExecute;
+
+    /// <inheritdoc/>
+    public event EventHandler CanExecuteChanged;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RelayCommand{T}"/> class that can always execute.
+    /// </summary>
+    /// <param name="execute">The execution logic.</param>
+    /// <remarks>
+    /// Due to the fact that the <see cref="System.Windows.Input.ICommand"/> interface exposes methods that accept a
+    /// nullable <see cref="object"/> parameter, it is recommended that if <typeparamref name="T"/> is a reference type,
+    /// you should always declare it as nullable, and to always perform checks within <paramref name="execute"/>.
+    /// </remarks>
+    /// <exception cref="System.ArgumentNullException">Thrown if <paramref name="execute"/> is <see langword="null"/>.</exception>
+    public RelayCommand(Action<T> execute)
+    {
+        if (execute is null)
+        {
+            throw new ArgumentNullException(nameof(execute));
+        }
+
+        _execute = execute;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RelayCommand{T}"/> class.
+    /// </summary>
+    /// <param name="execute">The execution logic.</param>
+    /// <param name="canExecute">The execution status logic.</param>
+    /// <remarks>See notes in <see cref="RelayCommand{T}(System.Action{T})"/>.</remarks>
+    /// <exception cref="System.ArgumentNullException">Thrown if <paramref name="execute"/> or <paramref name="canExecute"/> are <see langword="null"/>.</exception>
+    public RelayCommand(Action<T> execute, Predicate<T> canExecute)
+    {
+        if (execute is null)
+        {
+            throw new ArgumentNullException(nameof(execute));
+        }
+
+        if (canExecute is null)
+        {
+            throw new ArgumentNullException(nameof(canExecute));
+        }
+
+        _execute = execute;
+        _canExecute = canExecute;
+    }
+
+    /// <inheritdoc/>
+    public void NotifyCanExecuteChanged()
+    {
+        CanExecuteChanged.Invoke(this, EventArgs.Empty);
```

Comment:
This would throw if nobody is listening to CanExecuteChanged. Should be `CanExecuteChanged?.Invoke`.

Reply:
@zlatanov - Thanks for your suggestion, I have added this check and raised a PR for that - #8619.